perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_vo_attribute_def_def_membershipExpirationRules.java

```diff
@@ -164,17 +170,20 @@ private boolean isAllowedParameter(String parameter) {
 	 * @param date format of date (1.1. or 01.1. or 01.01. etc)
 	 * @return String standard format of date
 	 */
-	private String standardFormatDate(String date) {
-		int position1 = 0;
-		int position2 = 3;
-		if(date == null || date.length() == 0) return date;
-		if(date.charAt(position1) == '0') {
-			date = date.substring(position1+1);
-			position2--;
-		}
-		if(date.length() == (position2+3) && date.charAt(position2) == '0') {
-			date = date.substring(0,position2) + date.substring(position2+1);
+	private String standardFormatDate(String date) throws InternalErrorException {
+		if(date == null || date.isEmpty()) return date;
+		//remove all white spaces
+		date = date.replaceAll("\\s", "");
+
+		Matcher standardFormatDateMatcher = standardFormatDatePattern.matcher(date);
+
+		if(!standardFormatDateMatcher.matches()) {
+			throw new InternalErrorException("Date is not in supported format: " + date);
```

Comment:
Is it surely internal error? Doesn't it mean that the user tried to store bad value to attribute?